### PR TITLE
Add upper bounds to reverse dependencies of Jane Street packages

### PR DESCRIPTION
**Pull request by xclerc**
Follow-up to https://github.com/ocaml/opam-repository/pull/13010, https://github.com/ocaml/opam-repository/pull/13016, and https://github.com/ocaml/opam-repository/pull/13107.

### CODE REVIEW

**Comment by xclerc** [comment]
It looks like most problems are related to `ahrocksdb`,
and are apparently expected to be fixed by https://github.com/ocaml/opam-repository/pull/13347.

File: packages/biocaml/biocaml.0.9.0/opam [hunk 1]
```diff
@@ -29,20 +29,20 @@ depends: [
...
+  "ppx_compare" {< "v0.12"}
+  "ppx_sexp_conv" {< "v0.12"}
   "re"
   "rresult"
   "uri"
 ]
 depopts: [
-  "async"
-  "core"
+  "async" {< "v0.12"}
```
Comment (kit-ty-kate) [inline packages/biocaml/biocaml.0.9.0/opam]: this shouldn't be define like this if I'm right. It has a different semantic than `conflicts: ["async" {>= "v0.12"}]`

The current one becomes a dependency only if `async < v0.12` is installed but doesn't say anything for `async >= v0.12`

Comment (xclerc) [reply packages/biocaml/biocaml.0.9.0/opam]: Indeed, this is an embarrassing mistake; thanks for noticing!

Comment (kit-ty-kate) [reply packages/biocaml/biocaml.0.9.0/opam]: Mmh, the fix you just push is not correct. It's still more or less the same thing.
The constraint needs to be in the `conflicts` block only

Comment (xclerc) [reply packages/biocaml/biocaml.0.9.0/opam]: I was confused, the "fix" was definitely wrong.

**Comment by avsm** [comment]
thanks!

**Pull request merged**